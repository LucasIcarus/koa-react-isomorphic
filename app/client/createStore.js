/* @flow */
/* global process */
import { fromJS } from "immutable";
import { combineReducers } from "redux-immutable";
import { applyMiddleware, compose, createStore } from "redux";
import thunkMiddleware from "redux-thunk";
import { createLogger } from "redux-logger";
import injectReducers from "./helpers/injectReducers";
import reducers from "./createReducer";

const middlewares = [thunkMiddleware];
const enhancers = [];

// support for development
if (process.env.RUNTIME_ENV === "client") {
  if (process.env.NODE_ENV === "development") {
    middlewares.push(createLogger({
      level: "info",
      stateTransformer: state => state.toJS()
    }));
  }

  if (window.__REDUX_DEVTOOLS_EXTENSION__) {
    enhancers.push(global.__REDUX_DEVTOOLS_EXTENSION__());
  }
}

export default (initialState: Object = {}) => {
  const store = createStore(
    combineReducers(reducers),
    fromJS(initialState),
    compose(applyMiddleware(...middlewares), ...enhancers)
  );

  // enable async reducers for each page load
  // $FlowFixMe
  store.reducers = reducers;

  if (process.env.NODE_ENV === "development" && module.hot) {
    // $FlowFixMe
    module.hot.accept("./createReducer", () =>
      injectReducers(store, combineReducers(require("./createReducer").default))
    );
  }

  return store;
};
