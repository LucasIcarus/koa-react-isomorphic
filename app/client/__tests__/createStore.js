import createStore from "../createStore";

test("createStore default state", () => {
  expect(createStore().getState()).toMatchSnapshot();
});
