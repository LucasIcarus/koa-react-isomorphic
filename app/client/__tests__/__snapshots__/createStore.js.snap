// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStore default state 1`] = `
Object {
  "helmet": Object {
    "link": Array [],
    "title": "Koa React Isomorphic",
  },
  "routing": Object {
    "locationBeforeTransitions": null,
  },
}
`;
